pallets/Polkadex_Ido/src/lib.rs

```diff
@@ -521,6 +600,63 @@ decl_module! {
             Ok(())
         }
 
+        #[weight = 10000]
+        pub fn vote(origin, round_id: T::Hash, amount: T::Balance, vote_multiplier: u8, approve : bool) -> DispatchResult {
+            let current_block_no = <frame_system::Pallet<T>>::block_number();
+            ensure!(vote_multiplier <=  6,  Error::<T>::PeriodError);
+            let who: T::AccountId = ensure_signed(origin)?;
+            ensure!(<InfoFundingRound<T>>::contains_key(&round_id), Error::<T>::FundingRoundDoesNotExist);
+            let funding_round = <InfoFundingRound<T>>::get(&round_id);
+            ensure!(current_block_no < funding_round.vote_end_block , Error::<T>::VotingEnded);
+            let voting = <RoundVotes<T>>::get(&round_id);
+            let position_yes = voting.ayes.iter().position(|a| a.account_id == who);
+            let position_no = voting.nays.iter().position(|a| a.account_id == who);
+            // Detects first vote of the member in the motion
+            let is_account_voting_first_time = position_yes.is_none() && position_no.is_none();
+            ensure!(is_account_voting_first_time, Error::<T>::DuplicateVote);
```

Comment:
A second ballot should remove the first vote and update with a new, or is this not how voting works elsewhere here?

Reply:
Ok i think updating the ballot is a better design.